### PR TITLE
DocSearch

### PR DESCRIPTION
**Pull request by flying-sheep**
[rendered](https://scanpy.readthedocs.io/en/docsearch/)

/edit: huh! weird, seems that [readthedocs-sphinx-search](https://readthedocs-sphinx-search.readthedocs.io/en/latest/) pops up instead. why do we try to use docsearch instead of that one?

### CODE REVIEW

**Comment by ivirshup** [comment]
I added the readthedocssphinx search when experimenting with which one to use. DocSearch just seemed a lot nicer. It's got fuzzy matching, so using UK vs US spelling doesn't matter, also gives more context about results.

File: pyproject.toml [hunk 1]
```diff
@@ -90,7 +90,7 @@ doc = [
     'sphinx-rtd-theme>=0.3.1',
     'sphinx-autodoc-typehints',
     'readthedocs-sphinx-search',
```
Comment (ivirshup) [inline pyproject.toml]:
```suggestion
```

File: docs/conf.py [hunk 2]
```diff
@@ -94,7 +93,12 @@
 
 
 html_theme = 'scanpydoc'
-html_theme_options = dict(navigation_depth=4, logo_only=True)  # Only show the logo
+html_theme_options = dict(
+    navigation_depth=4,
+    logo_only=True,
+    docsearch_index='scanpy',
+    docsearch_key='fa4304eb95d2134997e3729553a674b2',
```
Comment (ivirshup) [inline docs/conf.py:100]: I think version needs to be passed, but when I try it locally I still seem to get duplicate results.

Comment (flying-sheep) [reply docs/conf.py:100]: It doesn’t, as documented here: https://icb-scanpydoc.readthedocs-hosted.com/en/latest/scanpydoc.theme.html#confval-docsearch_doc_version

Comment (flying-sheep) [reply docs/conf.py:100]: Must be something else. What’s “duplicated”? From different versions?

Comment (ivirshup) [reply docs/conf.py:100]: Yeah, I was getting results from both latest and stable.

**Review by ivirshup — APPROVED** [review]
Looks good now, must have been a cache issue. I'm good with merging, unless I'm forgetting something?

Release note would be good.

**Comment by flying-sheep** [comment]
Looking good! But somehow there’s like 3 times the same for each result … I assume two are expected as one is the page title and one the doc item title (e.g. function name), and we have one page per function/class.

But shouldn’t one side of the vertical line be e.g. the first line of the docstring? 

popup

**Pull request merged**